test/views/IndexViewSpec.scala

```diff
@@ -60,6 +79,20 @@ class IndexViewSpec extends SpecBase with ViewUtils with ViewBehaviours {
       }
     }
 
+    "display button with correct text and link when user owes penalties" when {
+      "user is trader" in {
+        val docWithPenalties = asDocument(applyView(userOwes = true))
+        docWithPenalties.select(Selectors.button).text() shouldBe whatYouOweButtonText
+        docWithPenalties.select(Selectors.button).attr("href").contains("http://localhost:9152/vat-through-software/what-you-owe") shouldBe true
+      }
+
+      "user is agent" in {
+        val docWithPenalties = asDocument(applyView(userOwes = true, isUserAgent = true))
+        docWithPenalties.select(Selectors.button).text() shouldBe whatYouOweButtonAgentText
+        docWithPenalties.select(Selectors.button).attr("href").contains("http://localhost:9152/vat-through-software/what-you-owe") shouldBe true
```

Comment:
Avoiding using shouldBe true
```suggestion
        docWithPenalties.select(Selectors.button).attr("href") shouldBe "http://localhost:9152/vat-through-software/what-you-owe"
```